crates/iceberg/src/scan.rs

```diff
@@ -383,18 +389,29 @@ mod tests {
 
             // prepare data
             let schema = {
-                let fields =
-                    vec![
-                        arrow_schema::Field::new("col", arrow_schema::DataType::Int64, true)
-                            .with_metadata(HashMap::from([(
-                                PARQUET_FIELD_ID_META_KEY.to_string(),
-                                "0".to_string(),
-                            )])),
```

Comment:
I took a look at the table metadata. The written Parquet schema doesn't match with the table schema.